devsetup/scripts/tripleo.sh

```diff
@@ -39,43 +41,88 @@ if [[ ! -f $SSH_KEY_FILE ]]; then
     exit 1
 fi
 
+sudo dnf install -y python-jinja2
 source ${SCRIPTPATH}/common.sh
 
+# Clock synchronization is important for both Ceph and OpenStack services, so both ceph deploy and tripleo deploy commands will make use of chrony to ensure the clock is properly in sync.
+# We'll use the NTP_SERVER environmental variable to define the NTP server to use, e.g.:
+# export NTP_SERVER=pool.ntp.org
+
+if [ $EDPM_COMPUTE_CELLS -eq 2 ] || [ $EDPM_COMPUTE_CELLS -gt 3 ] || [ $EDPM_COMPUTE_CELLS -eq 0 ] ; then
+    echo "Only a main cell0 plus a 2 additional compute cells supported yet!"
+    exit 1
+fi
+
+# TODO(bogdando): multi-cell with ceph HCI
+if [ $EDPM_COMPUTE_CELLS -eq 3 ] && [[ "$EDPM_COMPUTE_CEPH_ENABLED" == "true" ]]; then
+    echo "Ceph storage is not supported for multi-cell setup yet!"
+    exit 1
+fi
+
 if [[ -e /run/systemd/resolve/resolv.conf ]]; then
     HOST_PRIMARY_RESOLV_CONF_ENTRY=$(cat /run/systemd/resolve/resolv.conf | grep ^nameserver | grep -v "${EDPM_COMPUTE_NETWORK_IP%.*}" | head -n1 | cut -d' ' -f2)
 else
     HOST_PRIMARY_RESOLV_CONF_ENTRY=${HOST_PRIMARY_RESOLV_CONF_ENTRY:-$GATEWAY}
 fi
 
-if [[ ! -f $CMDS_FILE ]]; then
-    cat <<EOF > $CMDS_FILE
+set +x
+cat <<EOF > $MY_TMP_DIR/.standalone_env_file
+export RH_REGISTRY_USER="$REGISTRY_USER"
+export RH_REGISTRY_PWD="$RH_REGISTRY_PWD"
+EOF
+chmod 0600 $MY_TMP_DIR/.standalone_env_file
+
+cat <<EOF > $CMDS_FILE
+. \$HOME/.standalone_env_file
+
 set -ex
 sudo dnf install -y podman python3-tripleoclient util-linux lvm2
 
 sudo hostnamectl set-hostname undercloud.localdomain
 sudo hostnamectl set-hostname undercloud.localdomain --transient
 
+cat >\$HOME/nova_noceph.yaml <<__EOF__
```

Comment:
why don't we just add this file into the rest of the env files instead of creating like that
i mean, what is special about this one?

Reply:
it is a recommended approach in general to provide separate files for different configuration areas, rather than mangling large files.

Reply:
i meant more like, we are already carrying a bunch of environment files. Fine if you don't want to put this into an existing one but you can just add a new file under the tripleo directory. 
With this we have some environments/parameter_defaults coming from files, and others like this one being created by heredoc. I think it just further adds to the complexity of this 'tool'.